### PR TITLE
Add offsets_begin/end() to strings_column_view

### LINKED ISSUE #9548: [FEA] Add `offsets_begin()` and `chars_begin()` iterator accesors for string_column_view
**Is your feature request related to a problem? Please describe.**

If I want to access the offsets/chars of a strings column, I am required to know/assume the underlying type, e.g.,
```
strings_column_view s;
auto offsets = s.offsets().data<int32_t>();
auto chars = s.chars().data<int8_t>();
```

This is inconvenient and brittle as what if the underlying type changed?

**Describe the solution you'd like**

Add member functions to `string_column_view` that give me a typed iterator to the underlying offsets/chars w/o me having to specify the type.

I want to be able to write:
```
strings_column_view s;
auto offsets = s.offsets_begin();
auto chars = s.chars_begin();
```

This has an added advantage of being able to implicitly handle sliced string columns as the `offsets_begin()` and `chars_begin()` functions could return transform iterators that automatically account for the `offset` of the parent. 

Note that we already do something like this for [`lists_column_view`](https://github.com/rapidsai/cudf/blob/baf8275ffaecb756092dc2c8819602cd2cff9396/cpp/include/cudf/lists/lists_column_view.hpp#L99) and [`tdigest_column_view` ](https://github.com/rapidsai/cudf/blob/baf8275ffaecb756092dc2c8819602cd2cff9396/cpp/include/cudf/tdigest/tdigest_column_view.cuh#L106).

### PR DESCRIPTION
**Pull request by davidwendt**
Closes #9548 

Adds `offsets_begin()` and `offsets_end()` to `cudf::strings_column_view`. Also adds `chars_begin()` and `chars_end()` as well. The `offsets_begin()` and `offsets_end()` apply any sliced values (offset, size) from the parent.

Also changed a few source files where these new functions can be used.

### CODE REVIEW

File: cpp/include/cudf/strings/strings_column_view.hpp [hunk 1]
```diff
@@ -78,6 +99,25 @@ class strings_column_view : private column_view {
    * view  (i.e.: non-zero offset or reduced row count).
    */
   size_type chars_size() const noexcept;
+
+  /**
+   * @brief Return an iterator for the chars child column.
+   *
+   * This does not applies the offset of the parent.
+   * The offsets child must be used to properly address
```
Comment (jrhemstad) [inline cpp/include/cudf/strings/strings_column_view.hpp:107]: Maybe add an example, e.g.,
```
To access the first character of string `i` (accounting for a sliced column offset):
chars_begin() + *(offsets_begin() + i)
```

**Review by jrhemstad — COMMENTED** [review]
Any sanity checking tests we want to add?

**Comment by davidwendt** [comment]
> Any sanity checking tests we want to add?

Do you mean like `CUDF_EXPECTS` checks for valid values? Or do you mean a gtest?

**Review by vyasr — APPROVED** [review]
CMake approval.

File: cpp/include/cudf/strings/strings_column_view.hpp [hunk 2]
```diff
@@ -78,6 +99,28 @@ class strings_column_view : private column_view {
    * view  (i.e.: non-zero offset or reduced row count).
    */
   size_type chars_size() const noexcept;
+
+  /**
+   * @brief Return an iterator for the chars child column.
+   *
+   * This does not applies the offset of the parent.
```
Comment (harrism) [inline cpp/include/cudf/strings/strings_column_view.hpp:106]:
```suggestion
   * This does not apply the offset of the parent.
```

File: cpp/include/cudf/strings/strings_column_view.hpp [hunk 3]
```diff
@@ -78,6 +99,28 @@ class strings_column_view : private column_view {
...
+   * a sliced column offset) use: `chars_begin()[offsets_begin()[i]]`.
+   *
+   * @return Iterator pointing to the first char byte.
+   */
+  chars_iterator chars_begin() const;
+
+  /**
+   * @brief Return an end iterator for the offsets child column.
+   *
+   * This automatically applies the offset of the parent.
```
Comment (harrism) [inline cpp/include/cudf/strings/strings_column_view.hpp:119]: Why does chars_end apply the offset of the parent if `chars_begin()` does not?  I think maybe this is just a copy paste error in the docs?

**Review by harrism — CHANGES_REQUESTED** [review]
Couple of doc issues.

**Review by harrism — APPROVED** [review]

**Review by jrhemstad — APPROVED** [review]

**Comment by jrhemstad** [comment]
@gpucibot merge

**Comment by davidwendt** [comment]
rerun tests

**Pull request merged**